sql/core/src/main/scala/org/apache/spark/sql/execution/basicOperators.scala

```diff
@@ -306,6 +306,8 @@ case class UnsafeExternalSort(
   override def output: Seq[Attribute] = child.output
 
   override def outputOrdering: Seq[SortOrder] = sortOrder
+
+  override def outputsUnsafeRows: Boolean = true
```

Comment:
Should we also add an assertion to our set operations that the inputs are the same type of row?